### DISCUSSION TITLE
[Code with Codespaces] Course feedback survey!

### DISCUSSION THREAD

**Discussion by nguyenalex836** [skills/.github#550] [Code with Codespaces]
Thank you for completing this course! ✨ We are delighted to have you as a user, and are committed to providing the best possible experience through GitHub Skills. 

Please take a moment to answer as many (or as few) questions as possible regarding the course you have just completed.

1. What part of the course did you enjoy the most?
2. What level of experience do you have with GitHub?
3. What’s something new you learned that you didn’t know before?
4. What’s something about the course that you didn’t enjoy as much?
5. What would you like to learn more about regarding GitHub?
6. What are you planning to use GitHub for?
7. How likely are you to recommend this course to a friend or colleague?

Your feedback is invaluable as we endeavor to enhance GitHub Skills for all GitHub users 💛

**Comment by mjturtora** [comment]
_What part of the course did you enjoy the most?_  
    Once I got the hang of it.  

_What level of experience do you have with GitHub?_  
    On and off since 2015  

_What’s something new you learned that you didn’t know before?_  
    Codespaces  

_What’s something about the course that you didn’t enjoy as much?_    

- I'm thinking it would not be easy for a beginner. I'm at _least_ intermediate.  
- Had to delete cache to get it to work, that's fine, but then was stuck with a half-way created thing. Had to delete everything and start over - not too surprising!   
- Then was a bit mystified through the whole thing and still not sure what I did.   
- A second run-through will help.   
- Even better would be if the instructions included deleting each code space along the way. That would have made what was happening more clear. Eventually I had to anyway because I maxed them out by the last step or so and had to learn to delete them to finish.   
- Also, not sure about the "commit button" that was mentioned. Could have used command line but I just used the vsc extension to do the stuff and had to assume the warnings were ok and that it was putting it in the right place.  

_What would you like to learn more about regarding GitHub?_  
Actions.  

_What are you planning to use GitHub for?_
I use it to make money!  

_How likely are you to recommend this course to a friend or colleague?_  
0.75  But more because I mostly work alone. And that leads to a final comment. I read everything. I guess I get it. But still not sure how it's better than just using my local machine or a remote desktop. Perhaps explain that  little better? I guess it's good for collab but like I said, I don't get to do a lot of that.  

**Oh, and THANKS for the course. I'm sure it was a lot of work!**

**Reply by nguyenalex836** [reply]
@mjturtora Thank you so much for taking the time to write up this wonderful feedback! ✨ You've provided great content for our team to review - thanks again! 💛

**Comment by RafaelJohn9** [comment]
> Thank you for completing this course! ✨ We are delighted to have you as a user, and are committed to providing the best possible experience through GitHub Skills.

> Please take a moment to answer as many (or as few) questions as possible regarding the course you have just completed.
> 
> 1. What part of the course did you enjoy the most?
- Online coding space
- Simple and easy to follow guide containing visual elements

> 2. What level of experience do you have with GitHub?
- Second year

> 3. What’s something new you learned that you didn’t know before?
-  I didn't Github copilot gave multiple suggestions.

> 4. What’s something about the course that you didn’t enjoy as much?
-  I enjoyed the course, I didn't  disliked anything.

> 5. What would you like to learn more about regarding GitHub?
- Github Actions but I guess there is a course for that :)

> 6. What are you planning to use GitHub for?
- Store codes, version control, open source contributions, help in the GitHub community

> 7. How likely are you to recommend this course to a friend or colleague?
>  Am very likely to recommend this course to a colleague

> Your feedback is invaluable as we endeavor to enhance GitHub Skills for all GitHub users 💛

**Reply by cmiguelangel859-del** [reply]
Aja

**Comment by nshobert** [comment]
### What part of the course did you enjoy the most?
I really liked the auto-updating steps upon refresh. Good recommendation to have a couple browser tabs open, made it easy to follow along. 

### What level of experience do you have with GitHub?
academic use for solo dev for ~2 years, professional use for ~6 months

### What’s something about the course that you didn’t enjoy as much?
I liked the course. One tiny improvement would be to show "step 1 of 4", "step 2 of 4", etc. At one point I wondered how much was left.

### What’s something new you learned that you didn’t know before?
what codespaces would look and feel like, great introduction to it

### How likely are you to recommend this course to a friend or colleague
already sent link to a friend

**Comment by CynthiaWahome** [comment]
# Course Feedback

Thank you for the fantastic course! 
I am happy to share my thoughts and contribute to making the course even better.

## What part of the course did you enjoy the most?

I particularly enjoyed the hands-on tasks. They were easy to follow and engaging. The practical experience of building projects and using GitHub Actions for grading made the learning process enjoyable and rewarding.

## What level of experience do you have with GitHub?

I have been exploring GitHub for about 9 months, having opened my account in March this year. It’s been a valuable learning experience.

## What’s something new you learned that you didn’t know before?

I gained a solid understanding of GitHub Codespaces. Learning to create, customize, and manage development environments with Codespaces, along with the additional tasks such as setting up custom images and personalizing the workspace, enhanced my practical skills significantly.

## What’s something about the course that you didn’t enjoy as much?

The instructions on updating the `devcontainer.json` were clear initially, but I found it confusing whether the update applied to the previous or current workspace. Additionally, I was unsure about the implications of stopping codespaces and whether they could be reactivated or would be permanently lost. Perhaps, abit more clarity on these aspects would be helpful. 

## What would you like to learn more about regarding GitHub?

I am interested in exploring GitHub Copilot and learning more about advanced GitHub features. I look forward to continuing my GitHub journey and expanding my skills further.


## What are you planning to use GitHub for?

I plan to use GitHub primarily for version control as I transition into the tech field. This platform has become a valuable tool for my development and learning.

I plan to use GitHub primarily for version control as I transition into the tech field. This is all relatively new to me, but I’ve found a great playground here on GitHub. I’m committed to using it for the long haul and exploring all it has to offer. Cheers to new beginnings! 🎉

## How likely are you to recommend this course to a friend or colleague?

I would highly recommend this course, especially for beginners. It’s hands-on and provides a solid foundation for learning GitHub effectively.


Way to go!

**Comment by Maria-Leiliane** [comment]
De que parte do curso você mais gostou?
Conhecer melhor as funcionalidades do Enterprise

Qual é o seu nível de experiência com o GitHub?
Uso mais o Bitibucket mas atuação é só no processo de desenvolvimento dentro do git-flow

O que você aprendeu de novo e não sabia antes?
Sobre executores hospedados no GitHub ou executores auto-hospedados.

O que você não gostou tanto no curso?
Leitura extensa desnecessária o tempo dos módulos podem ser reduzidos com uma revisão de texto para tornar-lo mais claro, objetivo e didático.

O que você gostaria de aprender mais sobre o GitHub?
Pretendo ter conhecimento completo .

Para que você planeja usar o GitHub?
Para atuar com qualidade nos processos de desenvolvimento.

Qual a probabilidade de você recomendar este curso a um amigo ou colega? 8
Se melhorar o texto 10

**Reply by Doctorlove99** [reply]
Learning about GitHub.(2) Minimal (3)It covers planning, set up your( repository) that is store files, history revision and all associated information, all information  for a project.(4) It's interesting (5) I intend to have complete Knowledge (6) I intend to have complete knowledge  about GitHub.(&)Used for code sharing (9) It's an interested course to learn

**Reply by Thhe1oldlady** [reply]
looking for more info

**Comment by Elisabete-MO** [comment]
Sempre achei o github uma ferramenta incrivel e esse curso é só a introduçao de um universo de possibilidades. Achei o curso muito bom e me deixou com muita vontade de procurar por mais conhecimento de codespaces. Já indiquei para o meu marido fazer e com certeza indicarei pra qualquer dev q nao tenha esse conhecimento de codespace.
junior
toda a configuraçao possivel eu ainda nao conhecia

**Comment by clifapps** [comment]
## One of the better online step-by-step tutorials I've taken: no typo's; no out-of-date code, or instructions; even the grammar was correct!

### What part of the course did you enjoy the most?
Can't say; it was all good. 😄 

### What level of experience do you have with GitHub?
Limited experience beginner: I've done a few things here.

### What’s something new you learned that you didn’t know before?
Codespaces, obviously. Also, learned how to use comments in discussions, and using emoticons!

### What’s something about the course that you didn’t enjoy as much?
Everything was great!  😄 

### What would you like to learn more about regarding GitHub?
Anything related to using GitHub as an independent developer.

### What are you planning to use GitHub for?
Developing personal apps, and learning web development.

### How likely are you to recommend this course to a friend or colleague?
Very likely!

**Thank you for a well designed and informative tutorial!!**
